src/rdkafka_mock.c

```diff
@@ -1808,6 +1859,20 @@ rd_kafka_mock_coord_set(rd_kafka_mock_cluster_t *mcluster,
         return mcoord;
 }
 
+/**
+ * @brief Remove coordinator by broker id.
+ */
+void rd_kafka_mock_coord_remove(rd_kafka_mock_cluster_t *mcluster,
+                                int32_t broker_id) {
+        rd_kafka_mock_coord_t *mcoord, *tmp;
+
+        TAILQ_FOREACH_SAFE(mcoord, &mcluster->coords, link, tmp) {
+                if (mcoord->broker_id == broker_id) {
+                        rd_kafka_mock_coord_destroy(mcluster, mcoord);
```

Comment:
`break;` inside `if`.

Reply:
In the mock cluster you can have coordinators for multiple consumer groups and the coordinator can be the same broker for all those groups so it's possible there are more than one.